src/azure-cli-core/azure/cli/core/aaz/_field_value.py

```diff
@@ -444,3 +445,113 @@ def to_serialized_data(self, processor=None, keep_undefined_in_list=False,  # py
         if processor:
             result = processor(self._schema, result)
         return result
+
+
+class AAZIdentityObject(AAZObject):
+    M_USER_ASSIGNED = "miUserAssigned"
+    S_USER_ASSIGNED = "userAssigned"
+    M_SYSTEM_ASSIGNED = "miSystemAssigned"
+    S_SYSTEM_ASSIGNED = "systemAssigned"
+
+    def to_serialized_data(self, processor=None, **kwargs):
+        data = dict()  # parameters for build
+        if self._data == AAZUndefined:
+            result = AAZUndefined
+
+        elif self._data is None:
+            result = None
+
+        else:
+            result = {}
+            schema = self._schema
+
+            for name, field_schema in schema._fields.items():
+                if name in self._data:
+                    v = self[name].to_serialized_data(processor=processor, **kwargs)
+                    if v == AAZUndefined:
+                        continue
+
+                    if field_schema._serialized_name:
+                        name = field_schema._serialized_name
+
+                    if name in {
+                        self.M_SYSTEM_ASSIGNED, self.M_USER_ASSIGNED,
+                        self.S_SYSTEM_ASSIGNED, self.S_USER_ASSIGNED,
+                    }:
+                        data[name] = v
+                        if action := field_schema._flags.get("action", None):  # subcommand has action
+                            data["action"] = action
+
+                    else:
+                        result[name] = v
+
+        result = self.build_identity(data, result)
+
+        if not result:
+            result = {"type": "None"}  # empty identity
+
+        if not result and self._is_patch:
+            result = AAZUndefined
+
+        if processor:
+            result = processor(self._schema, result)
+
+        return result
+
+    def build_identity(self, data, result):
+        if not data:
+            return result
+
+        if action := data.get("action", None):  # subcommand
+            identities = set(result.pop("userAssignedIdentities", {}).keys())
+            has_system_identity = "systemassigned" in result.pop("type", "").lower()
+
+            user_assigned = data.get(self.S_USER_ASSIGNED, None)
+            system_assigned = data.get(self.S_SYSTEM_ASSIGNED, None)
+
+            if action == "remove":
+                if user_assigned is not None:
+                    if len(user_assigned) > 1:  # remove each
+                        identities -= set(user_assigned)
+
+                    else:  # remove all
+                        identities = {}
+
+                if identities:
+                    result["userAssignedIdentities"] = {k: {} for k in identities}
+                    if system_assigned or not has_system_identity:
+                        result["type"] = "UserAssigned"
+
+                    else:
+                        result["type"] = "SystemAssigned, UserAssigned"
```

Comment:
Please check whether the space after `,` can always work? You can find some swagger definition without space, For example https://github.com/Azure/azure-rest-api-specs/blob/b72e0199fa3242d64b0b49f38e71586066a8c048/specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-05-01/workspace.json#L1014-L1019

Reply:
after testing, it seems `"SystemAssigned,UserAssigned` always work. will change `"SystemAssigned, UserAssigned"` to `"SystemAssigned,UserAssigned"`